ambry-commons/src/main/java/com.github.ambry.commons/NettyMetrics.java

```diff
@@ -0,0 +1,196 @@
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicBoolean;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+
+/**
+ * A class to register and collect netty-related metrics.
+ * To use this metric, please do
+ * <p>
+ *   NettyMetrics nettyMetrics = new NettyMetric(new MetricRegistry(), nettyConfig);
+ *   nettyMetrics.start();
+ * </p>
+ *
+ * This would start collecting netty metrics.
+ *
+ * To Stop collecting, please do
+ * <p>
+ *   nettyMetrics.stop();
+ * </p>
+ */
+public class NettyMetrics {
+
+  private static final Logger logger = LoggerFactory.getLogger(NettyMetrics.class);
+
+  /**
+   * Constructor to create a {@link NettyMetrics};
+   * @param registry Registry to registry the metrics.
+   * @param config {@link NettyConfig}.
+   */
+  public NettyMetrics(MetricRegistry registry, NettyConfig config) {
+    Objects.requireNonNull(registry, "Registry is null");
+    Objects.requireNonNull(config, "Netty config is null");
+    this.registry = registry;
+    this.config = config;
+  }
+
+  private volatile MetricRegistry registry;
+  private volatile NettyConfig config;
+  private ScheduledExecutorService scheduler = null;
+  private AtomicBoolean started = new AtomicBoolean();
+
+  private volatile int numDirectArenas;
+  private volatile int numHeapArenas;
+  private volatile int numThreadLocalCaches;
+  private volatile long usedHeapMemory;
+  private volatile long usedDirectMemory;
+  private volatile long numHeapTotalAllocations;
+  private volatile long numHeapTotalDeallocations;
+  private volatile long numHeapTotalActiveAllocations;
+  private volatile long numDirectTotalAllocations;
+  private volatile long numDirectTotalDeallocations;
+  private volatile long numDirectTotalActiveAllocations;
+
+  private int getNumDirectArenas() {
+    return numDirectArenas;
+  }
+
+  private int getNumHeapArenas() {
+    return numHeapArenas;
+  }
+
+  private int getNumThreadLocalCaches() {
+    return numThreadLocalCaches;
+  }
+
+  private long getUsedHeapMemory() {
+    return usedHeapMemory;
+  }
+
+  private long getUsedDirectMemory() {
+    return usedDirectMemory;
+  }
+
+  private long getNumHeapTotalAllocations() {
+    return numHeapTotalAllocations;
+  }
+
+  private long getNumHeapTotalDeallocations() {
+    return numHeapTotalDeallocations;
+  }
+
+  private long getNumHeapTotalActiveAllocations() {
+    return numHeapTotalActiveAllocations;
+  }
+
+  private long getNumDirectTotalAllocations() {
+    return numDirectTotalAllocations;
+  }
+
+  private long getNumDirectTotalDeallocations() {
+    return numDirectTotalDeallocations;
+  }
+
+  private long getNumDirectTotalActiveAllocations() {
+    return numDirectTotalActiveAllocations;
+  }
+
+  private void register() {
+    registry.register(MetricRegistry.name(NettyMetrics.class, "NumberDirectArenas"),
+        (Gauge<Integer>) () -> getNumDirectArenas());
```

Comment:
optional: change following to method reference, just you did in run method : `PoolArenaMetric::numAllocations`

Reply:
Gauge is not a functional interface, so I can't use NettyMetrics::getXXXX.

Reply:
How about this? 
```
registry.register(MetricRegistry.name(NettyMetrics.class, "NumberDirectArenas"),
        (Gauge<Integer>) this::getNumDirectArenas);
```

Reply:
it worked, updated.